### PR TITLE
Ensured a user with no photo or about me story image can hibernate , closes #1139 , closes #1134

### LINKED ISSUE #1134: Orphaned pictures are still in the database.
Thanks for fixing this.
However, orphaned pictures are still available via deep link.
The delete function might work correctly, but pictures which have been uploaded in the past are still in the database.

_Originally posted by @marqqs in https://github.com/animina-dating/animina/issues/1093#issuecomment-2464720070_

### LINKED ISSUE #1139: Change the handling of deleting the last story
#1134 displayed a potential problem with the current logic.

I wanted to be sure that nor orphaned profiles without a single story are in the system but this leads to this problem. The only solution which I can come up with is that we allow users to delete even their about_you story (which is THE one story that should always be there) but auto change their visibility to hibernate (is that the name we use?) but only after we told the user that this is the consequence of remove the about_you story.

GUI wise we have to display it to a user when he/she is in hibernation. On every screen. In their face! And we have to make it easy to get back to normal mode again.

Please propose a better solution in this issue if you have an idea.

### PR DESCRIPTION
**Pull request by MICHAELMUNAVU83**
- I ensured if a user has been hibernated due to lack of profile photo  they see these flash message

https://github.com/user-attachments/assets/6c71506f-6b62-4d78-a897-713166f38e96

"You need to have a profile photo and a photo for your about me story to change your profile visibility." - If a user user has no profile photo and no about me story image

"You need to have a profile photo to change your profile visibility." -  If a user has no profile photo 

"You need to have a photo for your about me story to change your profile visibility." - If a user has no about me story image

### CODE REVIEW

File: lib/animina_web/components/profile/profile_components.ex [hunk 1]
```diff
@@ -571,7 +571,11 @@ defmodule AniminaWeb.ProfileComponents do
           "change_user_state"
         end
       }
-      data-confirm={"Are you sure you want to change from #{@value} to #{@state}?"}
+      data-confirm={
+        if @state != @value || @state != @similar_value do
```
Comment (wintermeyer) [inline lib/animina_web/components/profile/profile_components.ex:575]: We need a I18n version. And please use something like 

```
gettext("Are you sure you want to change from %{value} to %{state}?", 
  value: @value, 
  state: @state
)
```

File: lib/animina_web/live/profile_visibility.ex [hunk 2]
```diff
@@ -107,6 +104,92 @@ defmodule AniminaWeb.ProfileVisibilityLive do
     User.normalize(user)
   end
 
+  defp maybe_change_user_state("normalize", socket, nil, false) do
```
Comment (wintermeyer) [inline lib/animina_web/live/profile_visibility.ex]: Please DRY this code. Something like

```elixir
defp maybe_change_user_state("normalize", socket, profile_photo, about_me_photo) do
  cond do
    is_nil(profile_photo) and not about_me_photo ->
      {:noreply, show_error(socket, "You need to have a profile photo and a photo for your about me story to change your profile visibility.")}

    is_nil(profile_photo) and about_me_photo ->
      {:noreply, show_error(socket, "You need to have a profile photo to change your profile visibility.")}

    not about_me_photo ->
      {:noreply, show_error(socket, "You need to have a photo for your about me story to change your profile visibility.")}

    true ->
      handle_state_change("normalize", socket)
  end
end

defp maybe_change_user_state("hibernate", socket, _profile_photo, _about_me_photo) do
  handle_state_change("hibernate", socket)
end

defp show_error(socket, message) do
  socket
  |> put_flash(:error, with_locale(socket.assigns.language, fn -> gettext(message) end))
end

defp handle_state_change(action, socket) do
  case change_user_state(action, socket.assigns.current_user) do
    {:ok, user} ->
      {:noreply,
       socket
       |> assign(current_user: user)
       |> put_flash(
         :info,
         with_locale(socket.assigns.language, fn ->
           gettext("Profile visibility changed successfully.")
         end)
       )}

    {:error, _} ->
      {:noreply,
       socket
       |> put_flash(
         :error,
         with_locale(socket.assigns.language, fn ->
           gettext("Profile visibility change failed.")
         end)
       )}
  end
end
```

**Review by wintermeyer — CHANGES_REQUESTED** [review]

File: lib/animina_web/live/profile_visibility.ex [hunk 3]
```diff
@@ -107,6 +104,74 @@ defmodule AniminaWeb.ProfileVisibilityLive do
     User.normalize(user)
   end
 
+  defp maybe_change_user_state("normalize", socket, profile_photo, about_me_photo) do
+    cond do
+      is_nil(profile_photo) and not about_me_photo ->
```
Comment (wintermeyer) [inline lib/animina_web/live/profile_visibility.ex:109]: This cries for more refactoring. Too much duplication. Something like:

```elixir
defp maybe_change_user_state("normalize", socket, profile_photo, about_me_photo) do
  case {is_nil(profile_photo), not about_me_photo} do
    {true, true} ->
      show_visibility_error(socket, "You need to have a profile photo and a photo for your about me story to change your profile visibility.")

    {true, false} ->
      show_visibility_error(socket, "You need to have a profile photo to change your profile visibility.")

    {false, true} ->
      show_visibility_error(socket, "You need to have a photo for your about me story to change your profile visibility.")

    _ ->
      handle_state_change("normalize", socket)
  end
end

defp show_visibility_error(socket, message) do
  {:noreply,
   socket
   |> show_error(with_locale(socket.assigns.language, fn -> message end))}
end
```

**Review by wintermeyer — CHANGES_REQUESTED** [review]
Just to be sure: Do we have the policies in place to ensure that the state can not be changed in these circumstances? I want to be sure that a potential black hat hacker doesn't play around with our API or form to bypass this.

**Comment by MICHAELMUNAVU83** [comment]
I have added a policy for reactivating to ensure we check the user has a profile photo and has about me story with an image

**Review by wintermeyer — APPROVED** [review]

**Pull request merged**